resources/assets/components/UserOverview/index.js

```diff
@@ -1,149 +1,85 @@
 import React from 'react';
-import PropTypes from 'prop-types';
-import { map, keyBy } from 'lodash';
-import { RestApiClient } from '@dosomething/gateway';
+import gql from 'graphql-tag';
+import { map } from 'lodash';
+import { useQuery } from '@apollo/react-hooks';
 
 import Empty from '../Empty';
 import SignupCard from '../SignupCard';
 import MetaInformation from '../MetaInformation';
-import UserInformation from '../Users/UserInformation';
-
-class UserOverview extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.api = new RestApiClient();
-
-    this.state = {
-      loading: false,
-      signups: [],
-      campaigns: [],
-    };
-  }
-
-  componentDidMount() {
-    this.setState({
-      loading: true,
-    });
-
-    // Get user activity.
-    this.getUserActivity(this.props.user.id)
-      // Then get the campaign data tied to that activity.
-      .then(() => {
-        const ids = map(this.state.signups, 'campaign_id');
-        this.getCampaigns(ids);
-
-        this.setState({
-          loading: false,
-        });
-      });
+import UserInformation, {
+  UserInformationFragment,
+} from '../Users/UserInformation';
+
+const USER_OVERVIEW_QUERY = gql`
+  query UserOverviewQuery($id: String!) {
+    user(id: $id) {
+      ...UserInformation
+      permalink
+      source
+    }
+
+    signups(userId: $id, orderBy: "created_at,desc") {
+      id
+      quantity
+      whyParticipated
+
+      campaign {
+        id
+        internalTitle
+        startDate
+      }
+
+      posts {
+        id
+        type
+        url(w: 200, h: 200)
+      }
+    }
   }
 
-  /**
-   * Gets the user activity for the specified user and update state.
-   *
-   * @param {String} id
-   * @return {Object}
-   */
-  getUserActivity(id) {
-    const request = this.api.get('api/v3/signups', {
-      filter: {
-        northstar_id: id,
-      },
-      include: 'posts',
-      orderBy: 'created_at,desc',
-      limit: 80,
-    });
+  ${UserInformationFragment}
```

Comment:
Is this what allows it to use `...UserInformation` above?

Reply:
Yep, exactly! This follows a similar pattern we use in Phoenix to query data [via a central "loader" component](https://github.com/DoSomething/phoenix-next/blob/155df29f183c74c44e9710ab68fc7e137e7eb135/resources/assets/components/utilities/ContentfulEntryLoader/ContentfulEntryLoader.js#L21-L46) but colocate the actual "data requirements" with the components themselves (like [here](https://github.com/DoSomething/phoenix-next/blob/155df29f183c74c44e9710ab68fc7e137e7eb135/resources/assets/components/actions/PetitionSubmissioncAction/PetitionSubmissionAction.js#L22-L33)).